### PR TITLE
Local storage: save markdown/HTML, don't pollute Sentry

### PR DESCRIPTION
**Pull request by jimrandomh**
Makes it so that the code which saves unfinished posts and comments to localStorage works with all the editors (DraftJS, markdown, HTML) rather than just DraftJS. Also strips away a wrapper library around localStorage, and changes error handling so that browsers not having localStorage is only a `console.warn` rather than a `console.error` (because the wide variety of ways of reporting that there's no localStorage, was polluting Sentry).

### CODE REVIEW

**Comment by jimrandomh** [comment]
Basic tests performed: Type in a post editor, refresh, with DraftJS and then with Markdown. Do the same for a comment editor. Inspect the `localStorage` object in the console to see that the names assigned are non-conflicting. Submit a post to confirm that it gets cleared from localStorage.

Tests not yet performed: Other editable fields, such as sequence description, bio, etc.

**Comment by jimrandomh** [comment]
Fixes #1729.

File: packages/lesswrong/components/editor/EditorFormComponent.jsx [hunk 1]
```diff
@@ -175,24 +189,76 @@ class EditorFormComponent extends Component {
...
-      if (this.changeCount % 30 === 0) {
-        const rawContent = convertToRaw(newContent);
-        this.getStorageHandlers().set({state: rawContent, doc: document, name})
-      }
-    }
+    const changed = (currentContent != newContent);
     this.setState({draftJSValue: value})
+    
+    if (changed)
+      this.maybeSaveBackup();
```
Comment (Discordius) [inline packages/lesswrong/components/editor/EditorFormComponent.jsx]: Make this one-line, or add curly braces, in this case I feel uneasy about the lack of braces

File: packages/lesswrong/components/editor/EditorFormComponent.jsx [hunk 2]
```diff
@@ -175,24 +189,76 @@ class EditorFormComponent extends Component {
...
-      // Only save to localStorage on every 30th content change
-      // TODO: Consider debouncing rather than saving every 30th change
-      // TODO: Consider saving on blur
-      this.changeCount = this.changeCount + 1;
-      if (this.changeCount % 30 === 0) {
-        const rawContent = convertToRaw(newContent);
-        this.getStorageHandlers().set({state: rawContent, doc: document, name})
-      }
-    }
+    const changed = (currentContent != newContent);
```
Comment (Discordius) [inline packages/lesswrong/components/editor/EditorFormComponent.jsx]: `!==` not `!=`

**Comment by jimrandomh** [comment]
Fixed the half-finished refactor (oops), and re-did manual testing.

**Comment by Discordius** [comment]
Did you end up running your basic tests for other editable fields?

**Comment by jimrandomh** [comment]
`/sequencesnew` and `/newPost` have a name clash and share save-state, in prod. This patch neither fixes nor worsens it.

**Comment by jimrandomh** [comment]
Same for creating a new group or event.

**Pull request merged**